File: eng-task-grading-react-fe/src/model/student-view-dto.ts
import type { AttendanceDaySetRecordDto, AttendanceDto } from "./attendance-dto";
import type { CourseDto } from "./course-dto";
import type { GradeDto } from "./grade-dto";
import type { TaskDto } from "./task-dto";

export interface StudentViewLoginDto{
  studentNumber: string;
  captchaToken?: string; // Přidáno pole pro token z Turnstile
}

export interface StudentViewTokenDto {
  accessToken: string;
  refreshToken: string;
}

export interface StudentViewCourseDto{
  course: CourseDto;
  tasks: TaskDto[];
  attendances: AttendanceDto[];
  grades: GradeDto[];
  attendanceRecords: AttendanceDaySetRecordDto[];
}

File: EngTaskGradingNetBESolution/EngTaskGradingNetBE/Models/Dtos/StudentViewDto.cs
namespace EngTaskGradingNetBE.Models.Dtos
{
  public record StudentViewTokenDto(string AccessToken, string RefreshToken);

  public record StudentViewCourseDto(
    CourseDto Course, 
    List<TaskDto> Tasks, List<AttendanceDto> Attendances, 
    List<GradeDto> Grades, 
    List<AttendanceDaySetRecordDto> AttendanceRecords);
}
